### PR TITLE
[tests-only] added solar configurations for unit tests

### PR DESCRIPTION
**Pull request by kiranparajuli589**
With this PR:
- Add sonarcloud scanner to owncloud design system

Related issue:
- Fixes https://github.com/owncloud/owncloud-design-system/issues/1365

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+sonar.host.url=https://sonarcloud.io
+
+# =====================================================
+#   Meta-data for the project
+# =====================================================
+
+sonar.links.homepage=https://github.com/owncloud/owncloud-design-system
+sonar.links.ci=https://drone.owncloud.com/owncloud/owncloud-design-system/
+sonar.links.scm=https://github.com/owncloud/owncloud-design-system
+sonar.links.issue=https://github.com/owncloud/owncloud-design-systemissues
```
Comment (phil-davis) [inline sonar-project.properties:15]:
```suggestion
sonar.links.issue=https://github.com/owncloud/owncloud-design-system/issues
```

**Review by phil-davis — COMMENTED** [review]
The coverage https://sonarcloud.io/dashboard?id=owncloud_owncloud-design-system&pullRequest=1367 reports 0% - but I don't remember exactly how that works. It might update after the PR is merged.

When we merge this, we need to look and see if we need to adjust more.

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/owncloud-design-system/2974
The sonarcloud step is running straight away. It needs to wait until after the "unit tests" step is finished - @kiranparajuli589

File: .drone.star [hunk 2]
```diff
@@ -76,7 +90,13 @@ def testing(ctx):
                     'yarn test',
                 ],
                 'depends_on': ['build system']
-            }
+            },
+            {
+              "name": "sonarcloud",
+              "image": "sonarsource/sonar-scanner-cli:latest",
+              "pull": "always",
+              "environment": sonar_env
```
Comment (phil-davis) [inline .drone.star]: Looks like it needs:
```
'depends_on': ['unit tests']
```

Comment (kiranparajuli589) [reply .drone.star]: done!

**Comment by phil-davis** [comment]
https://sonarcloud.io/dashboard?id=owncloud_owncloud-design-system&pullRequest=1367
Coverage 35.7% :rocket:

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
https://sonarcloud.io/dashboard?id=owncloud_owncloud-design-system is now showing 35.7%  coverage, after the merge CI uploaded the analysis of `master` - good.

**Pull request merged**